src/test/kotlin/org/rust/RsTestBase.kt

```diff
@@ -99,6 +100,24 @@ abstract class RsTestBase : LightPlatformCodeInsightFixtureTestCase(), RsTestCas
             System.err.println("SKIP $name: $reason")
             return
         }
+
+        if (findAnnotationInstance<BothEditions>() != null &&
+            javaClass.getMethod(name).getAnnotation(MockEdition::class.java) == null) {
```

Comment:
What if I mark the whole test class with `MockEdition` annotation?
Maybe it's better to always ignore one of these annotations when we find both ones?

Reply:
There is a case in RsStdResolveTest where they are mixed. But :thinking: really we can just extract it to another file (like RsStdlibResolveTestEdition2015). And in this case, it looks better to forbid to implement both `BothEditions` and `MockEdition` (i.e. throw exception in this case).

Reply:
About `RsStdlibResolveTest`: in #3864 I drop code to launch tests with both editions to have exactly the same environment as the current stdlib